gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -0,0 +1,103 @@
+
+class ForwardResultsHandler:
+    """Calculate all results for each market slot for forward markets."""
+
+    def __init__(self, slot_length: Duration):
+        self.forward_market_enabled = True
+        self.slot_length = slot_length
+        self.orders: Dict[
+            str, Dict[int, Dict[DateTime, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.previous_device_stats = {}
+        self.current_device_stats: Dict[
+            int, Dict[DateTime, Dict[str, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.device_time_series: Dict[
+            int, Dict[int, Dict[DateTime, Dict[str, Dict]]]] = defaultdict(
+            lambda: defaultdict(lambda: defaultdict(dict)))
+        self._total_memory_utilization_kb = 0.0
+
+    # pylint: disable=unused-argument
+    def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
+        """
+        Update the forward market aggregated results with bids_offers_trades of current slot.
+        """
+        self._clear_device_statistics()
+        for area_uuid, area_result in core_stats.items():
+            if "forward_market_stats" not in area_result:
+                return
+
+            current_market_dt = from_format(current_market, DATE_TIME_FORMAT) if (
+                current_market := current_market_slot) else ""
+            for market_type_value, market_stats in area_result["forward_market_stats"].items():
+                market_type = int(market_type_value)
+                self._get_bids_offers_trades(area_uuid, market_type, market_stats)
+                current_results = handle_forward_results(current_market_dt, market_stats)
+                self._update_stats_and_time_series(current_results, market_type)
+        self._update_memory_utilization()
+
+    def update_from_repr(self, area_representation: Dict):
+        """
+        Updates the simulation results using area_representation data that arrive from the d3a-web.
+        """
+
+    @property
+    def all_db_results(self) -> Dict:
+        """Get dict with all the results in format that can be saved to the DB."""
+        results = {"orders": self.orders, "current_device_stats": self.current_device_stats,
+                   "device_time_series": self.device_time_series, "cumulative_net_energy_flow": {},
+                   "cumulative_market_fees": 0.}
+        return results
+
+    @property
+    def total_memory_utilization_kb(self):
+        """Get the total memory allocated by the results."""
+        return self._total_memory_utilization_kb
+
+    def _clear_device_statistics(self) -> None:
+        self.orders.clear()
+        self.current_device_stats.clear()
+        self.device_time_series.clear()
+
+    def _get_bids_offers_trades(self, area_uuid: str, market_type: int,
+                                market_stats: Dict[str, Dict]):
+        for time_slot, orders in market_stats.items():
+            time_slot_dt = from_format(time_slot, DATE_TIME_FORMAT)
```

Comment:
There are a lot of forth and backs in the code between pendulum and string. I wonder if you can all remove them because you are not using the pendulums functionalities AFAIS. How about just using the already provided time_slot strings from the DB or from the results. Please correct me, if I overlooked something. Thanks!